Clinical-Genomics/scout
# Simplify case report loading commands and add RNA multiqc report loading

- Load an RNA multiqc for a case manually with command `scout load report -t multiqc_rna` (fix #3647 and fix partially #3648) 
- Load RNA multiqc report automatically for a case if its config file contains the `multiqc_rna` key/value
- Simplifies loading of all reports by refactoring the associated code. Specifically:
   - Deprecating CLI commands `scout load <delivery_report, gene_fusion_report, coverage_qc_report, cnv_report>`
   - Replacing them with command: `scout load report -t <report type>`

BONUS feature:
- Update multiqc file for a case using the command `scout load report -t multiqc`

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>


**How to test -> locally**:
1. Try to **load/update the 5 existing types of reports with the usual commands `case load`**:
   - case load cnv-report
   - case load coverage-qc-report
   - case load delivery-report
   - case load gene-fusion-report (--research) # it's 2 distinct reports for clinical and research
  
All these command should continue to work but should raise a deprecation message

2. **Load/update the same reports as before using the new command: scout load report <report-type>**

All 5 reports should be loaded and visible on case page

3. **Try loading the RNA multiqc report (any HTML file will do) with the same command as in 2.**

This report should be also visible on case page sidebar

4. **Testing of bonus feature: try to load/update a multiqc report** (any HTML file will do) with the command: `scout load report -t multiqc`

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
